src/query_sink.rs

```diff
@@ -0,0 +1,60 @@
+use winapi::{
+    um::wbemcli::{
+        IWbemClassObject,
+        WBEM_NO_ERROR,
+    },
+    shared::{
+        ntdef::HRESULT,
+        wtypes::BSTR,
+    },
+    ctypes::{
+        c_long,
+    },
+};
+use winapi::um::wbemcli::{IWbemObjectSink, IWbemObjectSinkVtbl};
+use com_impl::{ComImpl, VTable, Refcount};
+
+/// Implementation for IWbemObjectSink.
+/// This [Sink] receives asynchronously the result of the query,
+/// through Indicate calls. When finished,the SetStatus method
+/// is called.
+/// [Sink]: https://en.wikipedia.org/wiki/Sink_(computing)
+/// # https://docs.microsoft.com/fr-fr/windows/win32/wmisdk/example--getting-wmi-data-from-the-local-computer-asynchronously
+#[repr(C)]
+#[derive(ComImpl)]
+#[interfaces(IWbemObjectSink)]
+pub struct QuerySink {
+    vtbl: VTable<IWbemObjectSinkVtbl>,
+    refcount: Refcount,
+}
+
+impl QuerySink {
+    pub fn new_ptr() -> *mut IWbemObjectSink {
```

Comment:
I think it will be nicer to have a new like in the example:
https://docs.rs/com-impl/0.1.1/com_impl/
`pub fn new(...) -> ComPtr<...>`

Reply:
I struggled a bit with ComPtr, I was returning a ComPtr at first but then... I lost my way ^^
I thought that using "new" as a name for the function was a bit misleading as it's returning a pointer, but others in the rust told me that it was idiomatic to work with pointer in the COM API, so i'll go with new here.